### PR TITLE
Replace magic string arguments with enums

### PR DESCRIPTION
**Pull request by jwodder**

### CODE REVIEW

File: dandi/tests/test_download.py [hunk 1]
```diff
@@ -88,7 +100,7 @@ def test_download_000027_metadata_only(url: str, tmp_path: Path) -> None:
 @mark.skipif_no_network
 @pytest.mark.parametrize(
     "url",
-    [  # Should go through API
+    [  # Should go through AP
```
Comment (yarikoptic) [inline dandi/tests/test_download.py:103]: what is AP?

**Comment by yarikoptic** [comment]
cool!

File: dandi/tests/test_download.py (hunk 1 above)
Comment (jwodder) [reply dandi/tests/test_download.py:103]: I did not mean to make that change, and I don't know how it happened.

Comment (yarikoptic) [reply dandi/tests/test_download.py:103]: ok, pushed 10d076e7bd9625840d8a4e1cab8a49d986c415ea directly to master to revert this one

**Pull request merged**